fetching/tarball-fetcher/src/remoteTarballFetcher.ts

```diff
@@ -157,27 +157,43 @@ export function createDownloader (
           ? opts.onProgress
           : undefined
         let downloaded = 0
+        const chunks: Buffer[] = []
         res.body!.on('data', (chunk: Buffer) => {
+          chunks.push(chunk)
           downloaded += chunk.length
           if (onProgress != null) onProgress(downloaded)
         })
 
         // eslint-disable-next-line no-async-promise-executor
         return await new Promise<FetchResult>(async (resolve, reject) => {
-          const stream = res.body!
+          res.body!
             .on('error', reject)
+          await new Promise<void>((resolve) => res.body!.on('end', () => {
```

Comment:
FYI, the node built-in module `node:events` contains the `once` helper that creates a promise resolved by an event firing.

Reply:
Thanks